libs/helm/sidebar/src/lib/hlm-sidebar-wrapper.ts

```diff
@@ -1,24 +1,29 @@
 import { Directive, ElementRef, computed, inject, input } from '@angular/core';
 import { hlm } from '@spartan-ng/brain/core';
-
 import type { ClassValue } from 'clsx';
+import { injectHlmSidebarConfig } from './hlm-sidebar.token';
 
 @Directive({
 	selector: 'div[hlmSidebarWrapper]', // Restrict to main elements only
-
 	host: {
 		'data-slot': 'sidebar-wrapper',
 		'[class]': '_computedClass()',
+		'[style.--sidebar-width]': 'sidebarWidth()',
+		'[style.--sidebar-width-icon]': 'sidebarWidthIcon()',
 	},
 })
 export class HlmSidebarWrapper {
 	private readonly _element = inject<ElementRef<HTMLElement>>(ElementRef);
+	private readonly _config = injectHlmSidebarConfig();
 
 	public readonly userClass = input<ClassValue>('', { alias: 'class' });
 	protected readonly _computedClass = computed(() =>
-		hlm('group/sidebar-wrapper has-data-[variant=inset]:bg-sidebar flex min-h-svh w-full', this.userClass()),
+		hlm('group/sidebar-wrapper has-[[data-variant=inset]]:bg-sidebar flex min-h-svh w-full', this.userClass()),
```

Comment:
Updated to support Tailwind CSS v3, may need to check other classes if they work in v3